conans/util/env_reader.py

```diff
@@ -6,11 +6,17 @@
         get_env('CONAN_SSL_ENABLED', False) => Will autotransform ENV CONAN_SSL_ENABLED to boolean
 
 """
-import os
 
 
-def get_env(env_key, default=None, environment=os.environ):
+def get_env(env_key, default=None, environment=None):
     """Get the env variable associated with env_key"""
+
+    if environment is None:
+        # Explicit import to avoid any problem caching environ from os module
+        # if import is declared outside. It's a problem launching tests in parallel
+        import os
```

Comment:
import shouldn't be related to this issue, this is not cached, I don't see the issue why this import is local

Reply:
@memsharded The issue happens in unit tests if import it outside. I though that it was better to add a comment to avoid a future refactor and move the import to the outside like it was before.

Reply:
Which issue?

Reply:
The issue is: integration test using this function ``get_env`` running with nose in Mac OS X python 3.6 not passing while executing all the tests in parallel. I moved the import to the inside of the function to avoid any future problems with tests. I thought it was worth a reference for this import inside the function.

Reply:
WTF? but we are importing ``os`` everywhere, do you know the root cause?

Reply:
So the issue is because of the ``import`` and not the ``os.environ`` as default argument? This is very surprising, I thought the ``os.environ`` was the cause, it really explained the issue, now I am very confused, no idea why this would happen.

Reply:
I'm investigating the differences between os.environ in python 2.7 and 3.6. There are some differences in the object itself and probably about concurrency ....

Python 3.7: 

```
class _Environ(MutableMapping):
    def __init__(self, data, encodekey, decodekey, encodevalue, decodevalue, putenv, unsetenv):
        self.encodekey = encodekey
        self.decodekey = decodekey
        self.encodevalue = encodevalue
        self.decodevalue = decodevalue
        self.putenv = putenv
        self.unsetenv = unsetenv
        self._data = data
```
python 2.7:
```
        class _Environ(UserDict.IterableUserDict):
```

Checking also  ```tools.environment_append``` from the tests what is the effect on the real ```os.environ``` object.

I'll let you both know if I can see the reason.

Reply:
Please, remove the import, unfortunately, we cannot spend much time debugging this. 
If the CI is not complaining it's ok for me. I know that your test-suite will fail in parallel, but I prefer you to skip this test in your pycharm configuration launcher than debug python implementations.

Reply:
ok, all done then!!! import is already remove in the header only.

Reply:
nope! remove in the body, put it in the module header, let your test fail in local.

Reply:
@lasote test is failing with concurrency and without concurrency, always failing in the CI for python 3.6 MAC OS X, which I believe you told me is running without concurrency. So inside the function is the only way that I had to make it pass in CI.

Reply:
Ok, please, push it with the import in the header I want to see a fresh trace failing.

Reply:
@lasote ok, reverting it and pushing. In my local environment I can see the following trace only when executing ```nosetests . ``` (all of them).

```
======================================================================
ERROR: Test get_env is available and working in conanfile
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pvicente/git/conan/conan/conans/test/util/tools_test.py", line 151, in test_get_env_in_conanfile
    client.run("build .")
  File "/Users/pvicente/git/conan/conan/conans/test/utils/tools.py", line 440, in run
    raise Exception("Command failed:\n%s" % command_line)
Exception: Command failed:
build .
-------------------- >> begin captured stdout << ---------------------
test_get_env_in_conafile CONAN_RUN_TESTS=False
Project: Running build()
ERROR: Hello/0.1@PROJECT: Error in build() method, line 10
	assert(run_tests == True)
	AssertionError:
```